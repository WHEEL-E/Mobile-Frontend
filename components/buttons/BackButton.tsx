import React from "react";
import { StyleSheet, TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";
import { InputBackButtonProps } from "../../utilities/buttonsUtils";

export const BackButton = (props: InputBackButtonProps) => {
  const { onPress } = props;

  return (
    <TouchableOpacity style={styles.button}>
      <Ionicons name="arrow-back" color="#3F414E" size={35} onPress={onPress} />
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    width: DEVICE_WIDTH * 0.15,
    height: DEVICE_WIDTH * 0.15,
    borderRadius: DEVICE_WIDTH * 0.075,
    alignItems: "center",
    justifyContent: "center",
    borderColor: "#EBEAEC",
    borderWidth: 0.7,
  },
});
