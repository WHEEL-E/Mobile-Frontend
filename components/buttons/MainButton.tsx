import React from "react";
import { StyleSheet, View, TouchableOpacity, Text, Image } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import { MainButtonProps } from "../../utilities/buttonsUtils";

export const MainButton = (props: MainButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress, icon, image } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <View style={styles.imageContainer}>
        {icon && (
          <Ionicons name={icon.name} size={icon.size} color={icon.color} />
        )}
        {image && <Image source={image.url} style={{ ...styles.image }} />}
      </View>
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
  },
  imageContainer: {
    width: "60%",
    height: "60%",
    justifyContent: "center",
    alignItems: "center",
  },
  image: {
    width: "100%",
    height: "100%",
    resizeMode: "center",
  },
});
