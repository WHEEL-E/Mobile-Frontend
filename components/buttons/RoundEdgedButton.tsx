import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { roundEdgedButtonProps } from "../../utilities/types/buttonTypes";
import { DEVICE_WIDTH } from "../../utilities/constants/dimentions";

const RoundEdgedButton = (props: roundEdgedButtonProps) => {
  const { title, backgroundColor, onPress } = props;
  return (
    <TouchableOpacity
      onPress={onPress}
      style={{ ...styles.button, backgroundColor }}
    >
      <Text style={styles.titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

export default RoundEdgedButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    height: 55,
    width: "80%",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 30,
    backgroundColor: "green",
    alignSelf: "center",
  },
  titleStyle: {
    fontFamily: "Cairo-Bold",
    color: "white",
    fontSize: DEVICE_WIDTH * 0.05,
  },
});
