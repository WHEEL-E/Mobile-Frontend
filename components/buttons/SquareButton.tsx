import React from "react";
import { StyleSheet, TouchableOpacity, Text } from "react-native";
import { SquareButtonProps } from "../../utilities/types/buttonTypes";

export const SquareButton = (props: SquareButtonProps) => {
  const { title, buttonStyle, titleStyle, onPress } = props;

  return (
    <TouchableOpacity
      style={{ ...styles.button, ...buttonStyle }}
      onPress={onPress}
    >
      <Text style={titleStyle}>{title}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    justifyContent: "center",
    alignItems: "center",
    borderWidth: 2,
    borderColor: "white",
    borderRadius: 10,
  },
});
