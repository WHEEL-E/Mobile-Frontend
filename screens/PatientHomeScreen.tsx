import React from "react";
import { Image, StyleSheet, Text, View } from "react-native";
import { useTranslation } from "react-i18next";
import colors from "../utilities/constants/colors";
import fonts from "../utilities/constants/fonts";
import { MainButton } from "../components/buttons/MainButton";
import { SquareButton } from "../components/buttons/SquareButton";
import { PatientHomeProps } from "../utilities/types/navigationTypes/mainNavigationTypes";
import { EmergencyCallModal } from "../components/homeScreenComponents/EmergencyCallModal";

const PatientHomeScreen = (props: PatientHomeProps) => {
  const [isModalVisible, setIsModalVisible] = React.useState(false);
  const { navigation } = props;
  const { t } = useTranslation();

  return (
    <View style={styles.container}>
      <EmergencyCallModal
        isModalVisible={isModalVisible}
        setIsModalVisible={setIsModalVisible}
      />
      <View>
        <Image
          style={styles.logo}
          source={require("../assets/images/logo-b-app.png")}
        />
        <Text style={styles.mainText}>
          {t("patientHomeScreen.morning")}user
        </Text>
        <Text style={styles.subText} testID="welcomeText">
          {t("patientHomeScreen.wishGoodDay")}
        </Text>
      </View>
      <View style={styles.buttons}>
        <View style={styles.buttonsCol}>
          <MainButton
            title={t("patientHomeScreen.map")}
            titleStyle={{ ...styles.bigButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.mainButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {}}
            image={{ url: require("../assets/images/map.png") }}
          />
          <SquareButton
            title={t("patientHomeScreen.emergencyCall")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkPink,
            }}
            onPress={() => {
              setIsModalVisible(true);
            }}
          />
          <SquareButton
            title={t("patientHomeScreen.freeDriving")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {
              navigation.navigate("FreeDrive");
            }}
          />
          <SquareButton
            title={t("patientHomeScreen.addSupervisor")}
            titleStyle={{ ...styles.smallButtonTitle, color: colors.darkBlue }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightPurple,
            }}
            onPress={() => {}}
          />
          <SquareButton
            title={t("patientHomeScreen.help")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightGreen,
            }}
            onPress={() => {}}
          />
        </View>
        <View style={styles.buttonsCol}>
          <MainButton
            title={t("patientHomeScreen.healthStatus")}
            titleStyle={{ ...styles.bigButtonTitle, color: colors.darkBlue }}
            buttonStyle={{
              ...styles.mainButton,
              backgroundColor: colors.lightPurple,
            }}
            onPress={() => {}}
            image={{ url: require("../assets/images/health-state.png") }}
          />
          <SquareButton
            title={t("patientHomeScreen.getChair")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.darkGreen,
            }}
            onPress={() => {}}
          />
          <SquareButton
            title={t("patientHomeScreen.reminders")}
            titleStyle={{ ...styles.smallButtonTitle, color: "white" }}
            buttonStyle={{
              ...styles.smallButton,
              backgroundColor: colors.lightGreen,
            }}
            onPress={() => {
              navigation.navigate("Reminders");
            }}
          />
          <MainButton
            title={t("patientHomeScreen.myProfile")}
            titleStyle={styles.profileTitle}
            buttonStyle={styles.myProfileButton}
            onPress={() => {
              navigation.navigate("Profile");
            }}
            icon={{
              name: "person-outline",
              size: 70,
              color: colors.lightBrown,
            }}
          />
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "white",
    paddingHorizontal: 20,
    paddingTop: "10%",
  },
  logo: {
    height: 40,
    resizeMode: "center",
    width: "70%",
    alignSelf: "center",
  },
  mainText: {
    fontFamily: fonts.CairoBold,
    fontSize: 30,
  },
  subText: {
    fontFamily: fonts.CairoMedium,
    color: colors.darkGrey,
    fontSize: 20,
  },
  buttons: {
    marginBottom: 100,
    flex: 1,
    flexDirection: "row",
  },
  buttonsCol: {
    flex: 1,
  },
  mainButton: {
    borderRadius: 30,
    justifyContent: "center",
    flex: 3,
    margin: 2,
  },
  smallButton: {
    flex: 1,
    backgroundColor: colors.darkGreen,
    margin: 2,
  },
  myProfileButton: {
    backgroundColor: colors.darkGreen,
    justifyContent: "center",
    flex: 2,
    borderRadius: 30,
    margin: 2,
  },
  bigButtonTitle: {
    fontFamily: fonts.CairoBold,
    fontSize: 30,
    lineHeight: 40,
  },
  smallButtonTitle: {
    fontFamily: fonts.CairoSemiBold,
    fontSize: 18,
  },
  profileTitle: {
    fontFamily: fonts.CairoSemiBold,
    fontSize: 20,
    color: "white",
  },
});

export default PatientHomeScreen;
