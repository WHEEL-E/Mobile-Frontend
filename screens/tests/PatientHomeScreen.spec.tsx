import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react-native";
import * as redux from "react-redux";
import PatientHomeScreen from "../PatientHomeScreen";
import * as mainButton from "../../components/buttons/MainButton";
import * as squareButton from "../../components/buttons/SquareButton";
import { MainButton } from "../../components/buttons/MainButton";
import { SquareButton } from "../../components/buttons/SquareButton";

jest.mock("react-i18next", () => ({
  ...jest.requireActual("react-i18next"),
  useTranslation: () => {
    return {
      t: () => "translated text",
    };
  },
}));

describe("patienthomeScreen", () => {
  beforeAll(() => {
    jest
      .spyOn(redux, "useSelector")
      .mockImplementation(() => "mockFirstName mockLastName");
  });
  it("makes sure screen matches snapShot", () => {
    const dummyProp: any = () => {};
    const patientHomeScreen = renderer
      .create(<PatientHomeScreen navigation={dummyProp} route={dummyProp} />)
      .toJSON();
    expect(patientHomeScreen).toMatchSnapshot();
  });

  it("makes sure button componenets are calledd", () => {
    jest.spyOn(mainButton, "MainButton");
    jest.spyOn(squareButton, "SquareButton");

    const dummyProp: any = () => {};
    render(<PatientHomeScreen navigation={dummyProp} route={dummyProp} />);
    expect(MainButton).toBeCalledTimes(3);
    expect(SquareButton).toBeCalledTimes(6);
  });

  it("makes sure text is rendered properly", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <PatientHomeScreen navigation={dummyProp} route={dummyProp} />
    );

    const welcomeText = getByTestId("welcomeText");
    expect(welcomeText.children).toContain("translated text");
  });

  it("makes sure first name is rendered", () => {
    const dummyProp: any = () => {};
    const { getByTestId } = render(
      <PatientHomeScreen navigation={dummyProp} route={dummyProp} />
    );

    const welcomeText = getByTestId("morningText");
    expect(welcomeText.children).toContain("mockFirstName");
    expect(welcomeText.children).not.toContain("mockLastName");
  });
});
