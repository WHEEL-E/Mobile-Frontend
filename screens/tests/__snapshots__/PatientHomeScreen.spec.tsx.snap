// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`patienthomeScreen makes sure screen matches snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "paddingHorizontal": 20,
      "paddingTop": "10%",
    }
  }
>
  <View>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "height": 40,
          "resizeMode": "center",
          "width": "70%",
        }
      }
    />
    <Text
      style={
        Object {
          "fontFamily": "Cairo-Bold",
          "fontSize": 30,
        }
      }
      testID="morningText"
    >
      translated text
      mockFirstName
    </Text>
    <Text
      style={
        Object {
          "color": "#8F9198",
          "fontFamily": "Cairo-Medium",
          "fontSize": 20,
        }
      }
      testID="welcomeText"
    >
      translated text
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "marginBottom": 100,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderRadius": 30,
            "flex": 3,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-Bold",
              "fontSize": 30,
              "lineHeight": 40,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#B91646",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7D9EA",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#19456B",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#16C79A",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E7D9EA",
            "borderRadius": 30,
            "flex": 3,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "height": "100%",
                "resizeMode": "center",
                "width": "100%",
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#19456B",
              "fontFamily": "Cairo-Bold",
              "fontSize": 30,
              "lineHeight": 40,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#16C79A",
            "borderColor": "white",
            "borderRadius": 10,
            "borderWidth": 2,
            "flex": 1,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 18,
            }
          }
        >
          translated text
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#11698E",
            "borderRadius": 30,
            "flex": 2,
            "justifyContent": "center",
            "margin": 2,
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "60%",
              "justifyContent": "center",
              "width": "60%",
            }
          }
        >
          <Text />
        </View>
        <Text
          style={
            Object {
              "color": "white",
              "fontFamily": "Cairo-SemiBold",
              "fontSize": 20,
            }
          }
        >
          translated text
        </Text>
      </View>
    </View>
  </View>
</View>
`;
